replication_timeout_test.go

```diff
@@ -217,6 +219,24 @@ func incompleteReplicationResponseFilter(msg *simplex.Message, _, _ simplex.Node
 	return true
 }
 
+// modifies the replication response to only send every other quorum round
+func removeFinalizationsFromReplicationResponses(msg *simplex.Message, _, _ simplex.NodeID) bool {
+	if msg.VerifiedReplicationResponse != nil || msg.ReplicationResponse != nil {
+		newData := make([]simplex.VerifiedQuorumRound, 0, len(msg.VerifiedReplicationResponse.Data))
+
+		for i := 0; i < len(msg.VerifiedReplicationResponse.Data); i++ {
+			qr := msg.VerifiedReplicationResponse.Data[i]
+			// to keep the request well-formed, we remove the finalization
```

Comment:
This test got me thinking - why do we even send both?

```
diff --git a/epoch.go b/epoch.go
index 10e54f0..6a65fec 100644
--- a/epoch.go
+++ b/epoch.go
@@ -2418,6 +2418,9 @@ func (e *Epoch) handleReplicationRequest(req *ReplicationRequest, from NodeID) e
                }
 
                data[i] = *quorumRound
+               if data[i].Finalization != nil {
+                       data[i].Notarization = nil
+               }
        }
 
        response.Data = data
```

With the above patch, the tests pass... shouldn't we only send the finalization if we have it? Why send the notarization too?